src/fhirtypes/StructureDefinition.ts

```diff
@@ -286,6 +287,10 @@ export class StructureDefinition {
     if (path.startsWith('snapshot') || path.startsWith('differential')) {
       throw new InvalidElementAccessError(path);
     }
+    const parentName = this.type || 'Resource';
```

Comment:
I don't think you need the `|| 'Resource'` part here. The `this` we are using in this context is a StructureDefinition. If you look at the documentation for StructureDefinition found here http://www.hl7.org/fhir/structuredefinition.html, you will see that type is 1..1:
![image](https://user-images.githubusercontent.com/52747882/84262536-cc4f3180-aaeb-11ea-9018-1a6444cbd485.png)

So we can safely assume `this.type` is going to exist, and directly compare `value !== this.type` on the line below.